### PR TITLE
[PM-15905] feat: Restructure build info output

### PR DESCRIPTION
**Pull request by vvolkgang**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-15905

## 📔 Objective

Restructure build info output copied from our About view, to reduce vertical space while providing the necessary info required for our team to troubleshoot. All information was retained except CI Run Number, which we can get by navigating to the build source. Compiler flags line is removed if empty and while at it, added the build variant.

New structure:
```
© Bitwarden Inc. 2015-2024

Version: 2024.6.0 (1)
📱 iPhone17,1 🍏 iOS 18.1 📦 Production
🧱 commit: bitwarden/ios/main@abc123
💻 build source: bitwarden/ios/actions/runs/123/attempts/123
🛠️ compiler flags: DEBUG_MENU
``` 

## 📸 Screenshots

| :arrow_left: Current | 🆕 CI Build|
|--------|--------|
| image | image | 

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by KatherineInCode** [comment]
🤔 It does look like the new version won't have some information, and require tapping through to view the run itself on device. Would it be valuable to do both, have a link to the run *and* still provide all that information for a quick glance?

**Comment by vvolkgang** [comment]
> 🤔 It does look like the new version won't have some information, and require tapping through to view the run itself on device. Would it be valuable to do both, have a link to the run _and_ still provide all that information for a quick glance?

We discussed this in private, it should convey the same information with the exception of CI Run Number.

**Review by fedemkr — APPROVED** [review]
LGTM 🎉

File: BitwardenShared/Core/Platform/Utilities/CIBuildInfo.swift [hunk 1]
```diff
@@ -4,5 +4,5 @@
 ///
 enum CIBuildInfo {
     /// Dictionary containing the info generated by the CI process.
-    static let info: [String: String] = [:]
+    static let info: KeyValuePairs<String, String> = [:]
```
Comment (KatherineInCode) [inline BitwardenShared/Core/Platform/Utilities/CIBuildInfo.swift:7]: 🤔 I guess we do actually want these in a consistent order each time, huh.

**Review by KatherineInCode — APPROVED** [review]

**Pull request merged**